mesh/mesh.hpp

```diff
@@ -1182,6 +1188,10 @@ class Mesh
    /// Return the indices and the orientations of all faces of element i.
    void GetElementFaces(int i, Array<int> &faces, Array<int> &ori) const;
 
+   /** @brief Returns the elements sharing a face with element @a elem,
+       including @a elem. */
+   void FindFaceNeighbors(const int elem, std::set<int> & nghb) const;
```

Comment:
Small nitpicks:

* The standard MFEM-style interface for this kind of function is to use `Array<int>` for the returned list of faces. You could use `Array::Unique` to deduplicate (or use a `std::set` in the implementation and convert to an `Array` at the end).
* I don't see a problem per-se with using `std::set` here, but I think consistency of the API has some value
* If we don't use `std::set`, we can remove `<set>` from the includes in this header.
* In cases like this where a type like `std::set` is used for an output parameter, I think it's better to just return the result by value: the intent is more clear from the function signature, it allows for better const-correctness, and it's basically free with copy elision/move semantics/RVO.

Reply:
Good suggestion, and it is better to settle this now, to avoid API changes. I changed it to return an `Array`.